src/app/cli/src/commands/repository_list_command.rs

```diff
@@ -31,68 +33,59 @@ impl RepositoryListCommand {
         }
     }
 
-    // TODO: support multiple format specifiers
-    fn print_machine_readable(&self) -> Result<(), CLIError> {
-        use std::io::Write;
+    fn record_batch(&self) -> Result<RecordBatch, InternalError> {
+        let mut col_name = Vec::new();
+        let mut col_url = Vec::new();
 
         let mut repos: Vec<_> = self.remote_repo_reg.get_all_repositories().collect();
         repos.sort();
-
-        let mut out = std::io::stdout();
-        writeln!(out, "Name,URL")?;
-
-        for name in repos {
-            let repo = self
-                .remote_repo_reg
-                .get_repository(&name)
-                .map_err(CLIError::failure)?;
-            writeln!(out, "{},\"{}\"", name, repo.url)?;
-        }
-        Ok(())
-    }
-
-    fn print_pretty(&self) -> Result<(), CLIError> {
-        use prettytable::*;
-
-        let mut repos: Vec<_> = self.remote_repo_reg.get_all_repositories().collect();
-        repos.sort();
-
-        let mut table = Table::new();
-        table.set_format(TableWriter::<Vec<u8>>::get_table_format());
-
-        table.set_titles(row![bc->"Name", bc->"URL"]);
-
         for name in &repos {
-            let repo = self
-                .remote_repo_reg
-                .get_repository(name)
-                .map_err(CLIError::failure)?;
-            table.add_row(Row::new(vec![
-                Cell::new(name),
-                Cell::new(repo.url.as_ref()),
-            ]));
+            let repo = self.remote_repo_reg.get_repository(name).int_err()?;
+            col_name.push(name.as_str());
+            col_url.push(repo.url.to_string());
         }
 
-        // Header doesn't render when there are no data rows in the table
-        if repos.is_empty() {
-            table.add_row(Row::new(vec![Cell::new(""), Cell::new("")]));
-        }
-
-        table.printstd();
-        Ok(())
+        self.records(vec![
+            Arc::new(StringArray::from(col_name)),
+            Arc::new(StringArray::from(col_url)),
+        ])
     }
 }
 
 #[async_trait::async_trait(?Send)]
 impl Command for RepositoryListCommand {
     async fn run(&mut self) -> Result<(), CLIError> {
         // TODO: replace with formatters
```

Comment:
Doesn't seem to be relevant anymore, does it?

Reply:
Good catch